components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -161,6 +218,8 @@ ext.gleanGenerateMetricsAPI = {
         args "$sourceOutputDir"
         args "-s"
         args "namespace=$fullNamespace"
+        // We intentionally don't let out the local metrics.yaml, even if another one will
+        // be extracted from the AAR file. The parser will just ignore this one, if not present.
         args "$projectDir/metrics.yaml"
```

Comment:
Include these two as `inputs file(...)`, so that Gradle dependencies are correct.

Reply:
The problem with it is that the files *can* be there, but it's not mandatory. If I add that directive the build will fail if the files are not there. We can have the following situations:

- no `metrics.yaml` at all (that's ok, Glean will only report the baseline ping with its core metrics);
- a local `metrics.yaml` and no `pings.yaml` (Glean will report the new metrics through its [built-in pings](https://mozilla.github.io/glean/book/user/pings/index.html));
- a local `metrics.yaml` and a local `pings.yaml` (Glean will report the new metrics as defined in the pings file);
- local files + AAR  `metrics.yaml` file (Glean will process all the 'metrics.yaml' and do whatever is needed)

Reply:
> The problem with it is that the files _can_ be there, but it's not mandatory. If I add that directive the build will fail if the files are not there.

Huh, I thought missing inputs were accepted by Gradle, but apparently not: see https://devsoap.com/optional-input-files-for-gradle/.

Alternatively, if you put these files in `glean/`, then you could add the directory as input instead, which does the right thing generally...

But this can be for future expansion.

Reply:
> 
> 
> > The problem with it is that the files _can_ be there, but it's not mandatory. If I add that directive the build will fail if the files are not there.
> 
> Huh, I thought missing inputs were accepted by Gradle, but apparently not: see https://devsoap.com/optional-input-files-for-gradle/.

I was quite surprised about this as well :(

> Alternatively, if you put these files in `glean/`, then you could add the directory as input instead, which does the right thing generally...

Yes, I'm afraid that doing this change right now would require changing all the consumers of Glean :) I'd open that can of worms some other day ;)